theories/common.elpi

```diff
@@ -178,6 +178,12 @@ type rmorphism-Z   rmorphism. % (int_of_Z _)%:~R
 
 % destructors
 
+pred sring->pzsring i:term, o:term.
+sring->pzsring U V :-
+	if (coq.locate-all "pzSemiRingType" [loc-gref PzSemiRingType| _ ])
+		(coq.elaborate-skeleton U (global PzSemiRingType) V ok)
+	(V = U).
+
```

Comment:
Calling this predicate every time we process a ring operator looks a bit costly. Isn't it possible to define compatibility notations (or constants) for `GRing.SemiRing.sort`, `GRing.ComSemiRing.sort`, `GRing.Ring.sort`, and `GRing.ComRing.sort` and replace their occurrences in `common.elpi`?

Reply:
Yes, Cyril raised the same issue. He figured that we could make the code more robust to the changes that are happening in mathcomp by precomputing the coercions that `coq.elaborate-skeleton` would insert, see https://github.com/math-comp/algebra-tactics/pull/108).